src/main/java/com/gluonhq/substrate/model/ClassPath.java

```diff
@@ -0,0 +1,102 @@
+
+/**
+ * Represents the classpath
+ * Consolidates all classpath manipulations
+ */
+public class ClassPath {
+
+    private final String classPath;
+
+    /**
+     * Creates the class path
+     * @param classPath standard java classpath, usually delimited with "pathSeparator"
+     */
+    public ClassPath(String classPath ) {
+        this.classPath = Objects.requireNonNull(classPath);
+    }
+
+    private Stream<String> asStream() {
+        return Stream.of(classPath.split(File.pathSeparator));
+    }
+
+    /**
+     * Returns whether any elements of this classpath match the provided
+     * predicate.
+     * @param predicate predicate to apply to elements of this classpath
+     * @return {@code true} if any elements of the stream match the provided
+     *  predicate, otherwise {@code false}
+     */
+    public boolean contains( Predicate<String> predicate ) {
+        Objects.requireNonNull(predicate);
+        return asStream().anyMatch( predicate);
+    }
+
+    /**
+     * Returns a list of strings consisting of the elements of this classpath that match
+     * the given predicate.
+     * @param predicate predicate to apply to elements of this classpath
+     * @return filtered list of elements of this classpath
+     */
+    public List<String> filter( Predicate<String> predicate ) {
+        Objects.requireNonNull(predicate);
+        return asStream().filter(predicate).collect(Collectors.toList());
+    }
+
+    /**
+     * Returns a List consisting of the results of applying the given
+     * function to the elements of this classpath.
+     *
+     * @param <T> The element type of the resulting List
+     * @param mapper a <a href="package-summary.html#NonInterference">non-interfering</a>,
+     *               <a href="package-summary.html#Statelessness">stateless</a>
+     *               function to apply to each element
+     * @return the list
+     */
+    public <T> List<T> mapToList( Function<String, T> mapper) {
+        Objects.requireNonNull(mapper);
+        return asStream().map(mapper).collect(Collectors.toList());
+    }
+
+    /**
+     * Returns a String classpath consisting of the results of applying the given
+     * function to the elements of this classpath.
+     *
+     * @param mapper a <a href="package-summary.html#NonInterference">non-interfering</a>,
+     *               <a href="package-summary.html#Statelessness">stateless</a>
+     *               function to apply to each element
+     * @return the string classpath
+     */
+    public String mapToString( Function<String, String> mapper) {
+        Objects.requireNonNull(mapper);
+        return asStream().map(mapper).collect(Collectors.joining(File.pathSeparator));
+    }
+
+    /**
+     * Returns a String classpath consisting of the results of applying the given
+     * function to the elements of this classpath. Tries to find javafx libraries and replace them with full path to this library
+     * within provided JavaFX SDK
```

Comment:
There is nothing JavaFX specific to this method. IIUC, this simply replace a set of given jar files with jar files that are found in a given directory, provided they are indeed present in that given directory.

Hence, I think the wording "fx" should be removed from the signature.

Reply:
Done